hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azure/NativeAzureFileSystem.java

```diff
@@ -1198,6 +1201,17 @@ public void setEncodedKey(String anEncodedKey) {
     private void restoreKey() throws IOException {
       store.rename(getEncodedKey(), getKey());
     }
+
+    /**
+     * Check for the stream being open.
+     * @throws IOException if the stream is closed.
+     */
+    private void checkOpen() throws IOException {
```

Comment:
We should check for stream closure for all the methods in my opinion, if we are going to check for some of them.  In particular, we should check for flush() which is inherited and hflush, hsync and hasCapabilities (also sync for branch-2).

Reply:
Also, a similar issue exists for NativeAzureFsInputStream.  I should add that I'm inclined not to fix this stuff, the reason being that it has been like this for a long time, I'm not aware of any related customer support issues, and there is risk in any change.  With that said, I will +1 the changes if the risk is low--you are fixing valid issues and thank you for that!

Reply:
After thinking about this, I have changed my mind and no longer think it is necessary to check for closure inside flush, hflush, hsync, or hasCapabilities.  These are no-ops when the buffers are empty, except for hasCapabilities which doesn't throw (so it doesn't even make sense for hasCapabilities to check).

Reply:
yes, some bits of code do call flush() after its closed, so its safest to no-op it.